conans/cli/commands/upload.py

```diff
@@ -0,0 +1,78 @@
+import json
+
+from conans.cli.api.conan_api import ConanAPIV2
+from conans.cli.api.model import UploadBundle
+from conans.cli.command import conan_command, COMMAND_GROUPS, OnceArgument
+from conans.client.userio import UserInput
+from conans.errors import ConanException
+
+
+@conan_command(group=COMMAND_GROUPS['creator'])
+def upload(conan_api: ConanAPIV2, parser, *args):
+    """
+    Uploads a recipe and binary packages to a remote.
+    By default, only the latest revisions matching the reference are uploaded, unles
+    is specified.
+    By default, if a recipe reference is specified, it will upload the latest revision of all the
+    binary packages, unles --only-recipe is specified.
+    """
+    _not_specified_ = object()
+
+    parser.add_argument('reference', help="Recipe reference or package reference, can contain * as "
+                                          "wildcard at any reference field. e.g: 'lib/*'.")
+    parser.add_argument('-p', '--package-query', default=None, action=OnceArgument,
+                        help="Only upload packages matching a specific query. e.g: os=Windows AND "
+                             "(arch=x86 OR compiler=gcc)")
+    # using required, we may want to pass this as a positional argument?
+    parser.add_argument("-r", "--remote", action=OnceArgument, required=True,
+                        help='Upload to this specific remote')
+    parser.add_argument("--only-recipe", action='store_true', default=False,
+                        help='Upload only the recipe')
+    parser.add_argument("--force", action='store_true', default=False,
+                        help='Ignore the missing fields in the scm attribute and override '
+                             'remote recipe and packages with local regardless of recipe date')
+    parser.add_argument("--check", action='store_true', default=False,
+                        help='Perform an integrity check, using the manifests, before upload')
+    parser.add_argument('-c', '--confirm', default=False, action='store_true',
+                        help='Upload all matching recipes without confirmation')
+    parser.add_argument('--retry', default=None, type=int, action=OnceArgument,
```

Comment:
Remove the retry/retry_wait, those are config items

Reply:
Args retry still here?

Reply:
Nope, they are not, refresh page